### DISCUSSION TITLE
Is slint suit for build a desktop IM product ?  (like whatapp or telegram)

### DISCUSSION THREAD

**Discussion by calvin2021y** [slint-ui/slint#3673] [Q&A]
I think video is hard.  

without video context,  will it support a super height chat room windows ? (show context dynamic when user scrolling the window)

**Comment by ogoffart** [comment]
I hope that Slint would work for a IM product.

I have seen users doing things that looks like it: https://github.com/Heng30/gossipbox  https://github.com/Heng30/chatbox https://github.com/Heng30/uibox

**Reply by calvin2021y** [reply]
Thanks for the tips. 

I try the projects you list here, and find the build release target is around 26 ~ 35 MB.  

I want to know the c++ version app will be smaller compare to rust ?    (for example if I build a c++ project,  what will be the total distribute size for download and install ?)  (I hope it less than 10M with other function like network and sqlite)

**Reply by tronical** [reply]
The size between Rust and C++ builds are similar. The final size depends on your operating system and (crucially) what assets your design is using and if you’re embedding them in the binary or not. I suggest to start with https://github.com/slint-ui/slint-cpp-template/tree/main and see what size you end up. There are tweaks to further change it, but it’s a good starting point.
